src/multi-selector/MultiSelector.jsx

```diff
@@ -49,28 +52,44 @@ class MultiSelector extends React.Component {
         };
     }
 
-    assignItems = async values => {
-        const selected = this.state.selected.concat(values);
-        this.setState({ selected });
-        this.props.onChange(selected);
+    getSelected = () => {
+        return this.isControlled ? this.props.selected : this.state.selected;
+    };
+
+    updateSelected = updateFn => {
+        const selected = this.getSelected();
+        const selectedUpdated = updateFn(selected);
+
+        if (this.isControlled) {
```

Comment:
Can be simplified to always setState and onChange. State will be ignored when controlled mode

Reply:
I considered it...  but now that I think of it, a change of state in React means a re-render. When in fact this.state.selected won't be used in render because it's controlled. So it's a bit weird. I'd leave it.

Reply:
Okay